lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -2100,14 +2101,28 @@ GLStatus TFGraphFunctionLowering::lowerSequenceRegion(SequenceSESERegion *r) {
   for (auto &child : r->getNodes()) {
     // The outputs for a sequence corresponds to the outputs of the last region
     // in the sequence. Hence, clear outputs for the current function if any.
-    getCurrentGraphFunction().outputs.clear();
+    // Do not clear the outputs if the next region is a function as the outputs
```

Comment:
this feels a little "brittle" (or at least non-obvious to read), because it's temporarily breaking some invariant here (that is, the intermediate outputs of some sub-region in a SequenceRegion should always get cleared away), and that invariant is later restored in TFGraphFunctionLowering::lowerFunctionRegion().

How about here we check: if it's SESERegionTree::Function, **move** the  `getCurrentGraphFunction().outputs` into `child`, so that we continue to uphold the invariant: `getCurrentGraphFunction().outputs` is always cleared here?